Engine/Helper/Utils.cpp

```diff
@@ -172,4 +164,116 @@ float4x4 Utils::GetTransform(const aiMatrix4x4& current_transform, float scale_f
 	math::Quat rotation(pRotation.x, pRotation.y, pRotation.z, pRotation.w);
 	math::float3 translation(pPosition.x, pPosition.y, pPosition.z);
 	return math::float4x4::FromTRS(translation, rotation, scale);
+}
+
+size_t Utils::GetImageType(const std::string& file_extension)
+{
+	size_t image_type;
+
+	if (file_extension == "bmp")
+	{
+		image_type = IL_BMP;
+	}
+	else if (file_extension == "cut")
+	{
+		image_type = IL_CUT;
+	}
+	else if (file_extension == "dds")
+	{
+		image_type = IL_DDS;
+	}
+	else if (file_extension == "gif")
+	{
+		image_type = IL_GIF;
+	}
+	else if (file_extension == "ico")
+	{
+		image_type = IL_ICO;
+	}
+	else if (file_extension == "jpg")
+	{
+		image_type = IL_JPG;
+	}
+	else if (file_extension == "lif")
+	{
+		image_type = IL_LIF;
+	}
+	else if (file_extension == "mng")
+	{
+		image_type = IL_MNG;
+	}
+	else if (file_extension == "pcd")
+	{
+		image_type = IL_PCD;
+	}
+	else if (file_extension == "pcx")
+	{
+		image_type = IL_PCX;
+	}
+	else if (file_extension == "pic")
+	{
+		image_type = IL_PIC;
+	}
+	else if (file_extension == "png")
+	{
+		image_type = IL_PNG;
+	}
+	else if (file_extension == "pbm")
+	{
+		image_type = IL_PNM;
+	}
+	else if (file_extension == "pgm")
+	{
+		image_type = IL_PNM;
+	}
+	else if (file_extension == "ppm")
+	{
+		image_type = IL_PNM;
+	}
+	else if (file_extension == "psd")
+	{
+		image_type = IL_PSD;
+	}
+	else if (file_extension == "psp")
+	{
+		image_type = IL_PSP;
+	}
+	else if (file_extension == "bw")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "rgb")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "rgba")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "sgi")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "tga")
+	{
+		image_type = IL_TGA;
+	}
+	else if (file_extension == "tif")
+	{
+		image_type = IL_TIF;
+	}
+	else if (file_extension == "tiff")
+	{
+		image_type = IL_TIF;
+	}
+	else if (file_extension == "jasc")
+	{
+		image_type = IL_JASC_PAL;
+	}
+	else
+	{
+		image_type = IL_TYPE_UNKNOWN;
+	}
```

Comment:
Maybe better a switch like so I think it is:
switch (file_extension)
{
  case file_extension ("bmp"):
    break;
  case file_extension ("cut"):
    break;
    ...
}

and I'm not sure, but I think some programs differ between jpg and jpeg, so maybe you can put it just in case ^^

Reply:
ok! I will change it :)

Reply:
I looked at it and for devil JPG and JPEG are the same, so no problem with it. For the switch probably is more readable but the function is so long that I'm too lazy to change it xD